android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -445,6 +445,20 @@ public void run() {
                     }
                 });
                 break;
+            case R.id.nav_share:
+                try {
+                    Intent shareIntent = new Intent();
+                    shareIntent.setAction(Intent.ACTION_SEND);
+                    shareIntent.setType("text/plain");
+                    shareIntent.putExtra(Intent.EXTRA_TEXT,
+                            String.format(getString(R.string.whatsapp_promo_msg_template),
+                                    String.format(getString(R.string.app_share_url),getPackageName())));
+                    startActivity(shareIntent);
+                }
+                catch (Exception e) {
```

Comment:
This is not the correct way to handle such events.
Rather, you should use `startActivityForResult` and check for the correct request code in `onActivityresult`.
It might be possible that the user chooses not to share the event and presses back button, you need to acknowledge that action and let the user know that the event wasn't shared and he/she can  retry it again by adding an action to the SnackBar.
Same can be done for a successful share too.